lib/elixir/pages/anti-patterns/code-anti-patterns.md

```diff
@@ -371,6 +371,135 @@ There are few known exceptions to this anti-pattern:
+      key == desired_key && value
+    end)
+  end
+end
+```
+
+```elixir
+# URL query string with the planned format - OK!
+iex> Extract.get_value("name=Lucas&university=UFMG&lab=ASERG", "name")
+"Lucas"
+# Unplanned URL query string format - Crash explaining the problem to the client!
+iex> Extract.get_value("name=Lucas&university=institution=UFMG&lab=ASERG", "university")
+** (MatchError) no match of right hand side value: ["university", "institution", "UFMG"]
+  extract.ex:7: anonymous fn/2 in Extract.get_value/2 # <= left hand: [key, value] pair
+iex> Extract.get_value("name=Lucas&university&lab=ASERG", "university")
+** (MatchError) no match of right hand side value: ["university"]
+  extract.ex:7: anonymous fn/2 in Extract.get_value/2 # <= left hand: [key, value] pair
+```
+
+Elixir and pattern matching promote an assertive style of programming where you handle the known cases. Once an unexpected scenario arises, you can decide to address it accordingly based on practical examples, or conclude the scenario is indeed invalid and the exception is the desired choice.
+
+`case/2` is another important construct in Elixir that help us write assertive code, by matching on specific patterns. For exmaple, if a function returns `{:ok, ...}` or `{:error, ...}`, prefer to explicitly match on both patterns:
+
+```elixir
+case some_function(arg) do
+  {:ok, value} -> # ...
+  {:error, _} -> # ...
+end
+```
+
+In particular, avoid matching solely on `_`, as shown below, as it is less clear in intent and it may hide bugs if `some_function/1` adds new return values in the future:
+
+```elixir
+case some_function(arg) do
+  {:ok, value} -> # ...
+  _ -> # ...
+end
+```
+
+## Non-assertive truthiness
+
+#### Problem
+
+Elixir provides the concept of truthiness: where `nil` and `false` are considered "falsy" and all other values are "truthy". Many constructs in the language, such as `&&/2`, `||/2`, and `!/1` handle truthy and falsy values. Using those operators is not an anti-pattern. However, using those operators when all operands are expected to be booleans, may be an anti-pattern.
+
+#### Example
+
+The simplest scenario this anti-pattern manifest is conditions, such as:
+
+```elixir
+if is_binary(name) && is_integer(age) do
+  # ...
+else
+  # ...
+end
+```
+
+Given both operands of `&&/2` are booleans, the code is more generic than necessary, and potentially unclear.
+
+#### Refactoring
+
+To remove this anti-pattern, we can replace `&&/2`, `||/2`, and `!/1` by `and/2`, `or/2`, and `not/1` respectively. The new operators assert at least their first argument is a boolean:
+
+```elixir
+if is_binary(name) or is_integer(age) do
+  # ...
+else
+  # ...
+end
+```
+
+A more assertive version of the code may also replace `if/2` by `case/2`, applying similar concepts to the ones presented in [Non-assertive patterns](#non-assertive-patterns):
+
+```elixir
+case is_binary(name) or is_integer(age) do
+  true -> # ...
+  false -> # ...
+end
+```
```

Comment:
Is it really something we should be promoting?

I don't really have an opinion, but I thought this might be controversial and I'm not even sure this style is used in the elixir codebase.
(also interestingly elixir-styler took the opposite choice of [rewriting these boolean `case`s as `if`s](https://github.com/adobe/elixir-styler/blob/main/test/style/single_node_test.exs#L295-L310))